core/src/main/scala/org/apache/spark/deploy/security/README.md

```diff
@@ -0,0 +1,239 @@
+# Delegation Token Handling In Spark
+
+This document aims to explain and demystify delegation tokens as they are used by Spark, since
+this topic is generally a huge source of confusion.
+
+
+## What are delegation tokens?
+
+Delegation tokens (DTs from now on) are authentication tokens used by some services to replace
+Kerberos service tokens. Many services in the Hadoop ecosystem have support for DTs, since they
+have two very desirable advantages over Kerberos tokens:
+
+* No need to distribute Kerberos credentials
+
+In a distributed application, distributing Kerberos credentials is tricky. Not all users have
+keytabs, and when they do, it's generally frowned upon to distribute them over the network as
+part of application data.
+
+DTs allow for a single place (e.g. the Spark driver) to require Kerberos credentials. That entity
+can then distribute the DTs to other parts of the distributed application (e.g. Spark executors),
+so they can authenticate to services.
+
+* A single token is used for authentication
+
+If Kerberos authentication were used, each client connection to a server would require a trip
+to the Key Distribution Center (KDC) and generation of a service ticket. In a distributed system,
```

Comment:
not a bit deal but I think we better stick to one way. " Key Distribution Center (KDC)" vs "TGT (ticket granting ticket)".

Reply:
I'll just revert to KDC. Readers of this document are expected to at least be a little familiar with Kerberos, so those terms should be familiar.